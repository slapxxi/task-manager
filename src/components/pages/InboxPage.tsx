import { Store, Tasks } from '@local/components';
import * as React from 'react';

function InboxPage() {
  return (
    <Store>
      {({ tasks, actions }) => (
        <div>
          <Tasks
            tasks={tasks}
            onChange={actions.updateTask}
            onDelete={actions.deleteTask}
          />
        </div>
      )}
    </Store>
  );
}

export default InboxPage;
