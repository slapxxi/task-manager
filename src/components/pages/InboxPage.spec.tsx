import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';
import InboxPage from './InboxPage';

jest.mock('@local/components/IconSystem/IconSystem');

it('renders', () => {
  const { container } = render(
    <Router>
      <InboxPage />
    </Router>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
