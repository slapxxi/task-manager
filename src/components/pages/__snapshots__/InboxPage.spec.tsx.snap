// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  class="inboxPage"
>
  <div
    class="toolbar"
  >
    <ul
      class="container"
    >
      <li
        class="item"
      >
        <a
          aria-current="page"
          class="link active"
          href="/"
        >
          <div
            name="inbox"
            size="20"
          >
            icon-
            inbox
          </div>
        </a>
      </li>
    </ul>
  </div>
  <header
    class="header"
  >
    <div
      name="inbox"
      size="30"
    >
      icon-
      inbox
    </div>
    <h1
      class="title"
    >
      Inbox
    </h1>
    <div
      class="menu"
      name="dots"
      size="20"
    >
      icon-
      dots
    </div>
  </header>
  <div
    class="content"
  >
    <p>
      Tasks are empty.
    </p>
    <footer
      class="footer"
    >
      <button
        class="button sc-bwzfXH gfRuPt"
      >
        Create Task
      </button>
    </footer>
  </div>
</div>
`;
